### ISSUE TITLE
[BUG] Doesn't work with multiple row groups

### ISSUE THREAD

**Issue by felipepessoto** [mukunku/ParquetViewer#27]
**Parquet Viewer Version**
2.2

**Where was the parquet file created?**
C#

**Sample File**
[Test.zip](https://github.com/mukunku/ParquetViewer/files/5681867/Test.zip)

**Describe the bug**
In UtilityMethods class this line contains a bug after the first call:

![image](https://user-images.githubusercontent.com/1336227/101959701-97c84980-3be4-11eb-8b4f-c79873ea6bde.png)

For example, if I have two row groups with 2 lines each.
At first call the if will be if (rowIndex=0 >= readRecords=2) - OK
But the next calls will be if (rowIndex=2 >= readRecords=2) and it will break. Unless the second row group is bigger than the first, but it is buggy anyway, since it will skip rows.

After fixing this issue, I also found another problem, where the row count is not respected after the first row group:

![image](https://user-images.githubusercontent.com/1336227/101959955-2046ea00-3be5-11eb-9c4e-587075354558.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/1336227/101960484-4c169f80-3be6-11eb-8c42-d4a44785ea52.png)

**Additional context**

Note: This tool relies on the [parquet-dotnet](https://github.com/elastacloud/parquet-dotnet) library for all the actual Parquet processing. So any issues where that library cannot process a parquet file will not be addressed by us. Please open a ticket on that library's repo to address such issues.

**Comment by mukunku** [comment]
Thanks for the contribution @felipepessoto , I've merged it to [v2.3](https://github.com/mukunku/ParquetViewer/releases/tag/v2.3). Sorry it took so long but it's hard to make time.

**Comment by felipepessoto** [comment]
Thanks @mukunku. Will you cherry pick it to main?

**Comment by mukunku** [comment]
There's one issue with your proposed changes, it doesn't calculate the total number of records in the file correctly. While working on it I realized the Thrift metadata already has the record count in it so I made some additional changes to utilize that.

It's possible to cherry pick both commits to master but I want to make sure my change won't also break something, hence the beta release. After a few weeks I'll merge all of it to master.
